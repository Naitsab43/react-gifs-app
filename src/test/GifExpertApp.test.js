const { shallow } = require("enzyme");
import { GifExpertApp } from "../GifExpertApp";
import React from "react";
import "@testing-library/jest-dom";


describe('Pruebas a <GifExpertApp />', () => {
  
  test('Debe de crear el componente', () => {
    
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();

  })

  test('Debe de mostrar una lista de categorias', () => {
    
    const categories = ["Naruto", "Nanatsu no taizai"]

    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);


  })
  
  

})
