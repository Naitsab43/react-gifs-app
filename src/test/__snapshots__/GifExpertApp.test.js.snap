// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifExpertApp /> Debe de crear el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas a <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Nanatsu no taizai"
      key="Nanatsu no taizai"
    />
  </ol>
</Fragment>
`;
